From 841fc919aad32a4d2c5a23826ed71e4759c991fc Mon Sep 17 00:00:00 2001
From: Andrew Regenscheid <aregenscheid@gmail.com>
Date: Fri, 23 Oct 2026 01:41:46 -0400
Subject: [PATCH 1/2] Patched for Raspberry Pi 5

Build the kernel from raspberrypi/linux rpi-6.18.y (a72a0085, 6.18.29) with a config ported from the release-1.13 patch onto release-1.14 (2ec10eb) and normalized with the release-1.14 toolchain (clang 22.1.8).
---
 Pkgfile                   |   8 +-
 kernel/build/config-arm64 | 483 +++++++++++++++++++++++---------------
 kernel/prepare/pkg.yaml   |   6 +-
 3 files changed, 300 insertions(+), 197 deletions(-)

diff --git a/Pkgfile b/Pkgfile
index 81baabb..b53453c 100644
--- a/Pkgfile
+++ b/Pkgfile
@@ -103,10 +103,10 @@ vars:
   kspp_sha256: 0dc61ebf01f0dd1d0d2204432235fb2470bcf2ec020172c30a9d1c14f103f032
   kspp_sha512: 2b40c41ed0347a339053aef61c838b5131dd583fb0d1ac8f3361040361ba868724ad16dfdc13023ff2b0f51f32c0aba605316d5d76f9fc6c1d50d90b4ceedead
 
-  # renovate: datasource=git-tags extractVersion=^v(?<version>.*)$ depName=git://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git
-  linux_version: 6.18.57
-  linux_sha256: 5bd9acdc82ec5efc3790ae840e5aaa40dfde7efff261cf6dde902dc570acf98c
-  linux_sha512: 6d9b39bee05daf05849c0bacc7bd407b13f34882bd06f46dc8c69835e98aba762d41f81852b5af34728c5cf15832fecb047c01d182bf3b581e82921885653689
+  # rpi-6.18.y branch, commit a72a0085aeac (2026-05-13), kernel 6.18.29
+  linux_version: a72a0085aeac23ac8aa76c90c476e20626046ea7
+  linux_sha256: 88c2c60538e4c79d59d5aaeed8925cf9a12ec4550345c71d1458984f3942eb94
+  linux_sha512: 00277666941c8bd3d147e12d963c936681d66258c8767179f4052fa49bc9860df0bfc6535d954dae46e4e84ea1de01010c916123f3211b0c2e1f1ebe9e7f9b91
 
   # renovate: datasource=git-tags extractVersion=^libaio-(?<version>.*)$ depName=https://pagure.io/libaio.git
   libaio_version: 0.3.113
diff --git a/kernel/build/config-arm64 b/kernel/build/config-arm64
index a664ba9..754418f 100644
--- a/kernel/build/config-arm64
+++ b/kernel/build/config-arm64
@@ -1,6 +1,6 @@
 #
 # Automatically generated file; DO NOT EDIT.
-# Linux/arm64 6.18.57 Kernel Configuration
+# Linux/arm64 6.18.29 Kernel Configuration
 #
 CONFIG_CC_VERSION_TEXT="clang version 22.1.8"
 CONFIG_GCC_VERSION=0
@@ -21,7 +21,6 @@ CONFIG_CC_HAS_ASSUME=y
 CONFIG_CC_HAS_NO_PROFILE_FN_ATTR=y
 CONFIG_CC_HAS_COUNTED_BY=y
 CONFIG_CC_HAS_MULTIDIMENSIONAL_NONSTRING=y
-CONFIG_CC_OPT_INLINE_MEMSET="-mllvm -max-store-memset=4294967294"
 CONFIG_LD_CAN_USE_KEEP_IN_OVERLAY=y
 CONFIG_PAHOLE_VERSION=131
 CONFIG_IRQ_WORK=y
@@ -295,6 +294,7 @@ CONFIG_TRACEPOINTS=y
 CONFIG_CRASH_RESERVE=y
 CONFIG_VMCORE_INFO=y
 CONFIG_KEXEC_CORE=y
+CONFIG_HAVE_IMA_KEXEC=y
 # CONFIG_KEXEC is not set
 CONFIG_KEXEC_FILE=y
 CONFIG_KEXEC_SIG=y
@@ -309,11 +309,11 @@ CONFIG_ARM64=y
 CONFIG_CLANG_SUPPORTS_DYNAMIC_FTRACE_WITH_ARGS=y
 CONFIG_64BIT=y
 CONFIG_MMU=y
-CONFIG_ARM64_CONT_PTE_SHIFT=4
-CONFIG_ARM64_CONT_PMD_SHIFT=4
-CONFIG_ARCH_MMAP_RND_BITS_MIN=18
-CONFIG_ARCH_MMAP_RND_BITS_MAX=33
-CONFIG_ARCH_MMAP_RND_COMPAT_BITS_MIN=11
+CONFIG_ARM64_CONT_PTE_SHIFT=7
+CONFIG_ARM64_CONT_PMD_SHIFT=5
+CONFIG_ARCH_MMAP_RND_BITS_MIN=16
+CONFIG_ARCH_MMAP_RND_BITS_MAX=31
+CONFIG_ARCH_MMAP_RND_COMPAT_BITS_MIN=9
 CONFIG_ARCH_MMAP_RND_COMPAT_BITS_MAX=16
 CONFIG_STACKTRACE_SUPPORT=y
 CONFIG_ILLEGAL_POINTER_VALUE=0xdead000000000000
@@ -431,16 +431,13 @@ CONFIG_ARM64_WORKAROUND_TSB_FLUSH_FAILURE=y
 CONFIG_ARM64_ERRATUM_2054223=y
 CONFIG_ARM64_ERRATUM_2067961=y
 CONFIG_ARM64_ERRATUM_2441009=y
-CONFIG_ARM64_WORKAROUND_BROKEN_AMU_CONSTCNT=y
 CONFIG_ARM64_ERRATUM_2457168=y
 CONFIG_ARM64_ERRATUM_2645198=y
-CONFIG_ARM64_ERRATUM_3821522=y
 CONFIG_ARM64_WORKAROUND_SPECULATIVE_UNPRIV_LOAD=y
 CONFIG_ARM64_ERRATUM_2966298=y
 CONFIG_ARM64_ERRATUM_3117295=y
 CONFIG_ARM64_ERRATUM_3194386=y
 CONFIG_ARM64_ERRATUM_4193714=y
-CONFIG_ARM64_ERRATUM_4118414=y
 CONFIG_CAVIUM_ERRATUM_22375=y
 CONFIG_CAVIUM_ERRATUM_23144=y
 CONFIG_CAVIUM_ERRATUM_23154=y
@@ -460,10 +457,11 @@ CONFIG_ROCKCHIP_ERRATUM_3588001=y
 CONFIG_SOCIONEXT_SYNQUACER_PREITS=y
 # end of ARM errata workarounds via the alternatives framework
 
-CONFIG_ARM64_4K_PAGES=y
-# CONFIG_ARM64_16K_PAGES is not set
+# CONFIG_ARM64_4K_PAGES is not set
+CONFIG_ARM64_16K_PAGES=y
 # CONFIG_ARM64_64K_PAGES is not set
-# CONFIG_ARM64_VA_BITS_39 is not set
+# CONFIG_ARM64_VA_BITS_36 is not set
+# CONFIG_ARM64_VA_BITS_47 is not set
 CONFIG_ARM64_VA_BITS_48=y
 # CONFIG_ARM64_VA_BITS_52 is not set
 CONFIG_ARM64_VA_BITS=48
@@ -498,7 +496,7 @@ CONFIG_ARCH_HAS_GENERIC_CRASHKERNEL_RESERVATION=y
 CONFIG_TRANS_TABLE=y
 CONFIG_XEN_DOM0=y
 CONFIG_XEN=y
-CONFIG_ARCH_FORCE_MAX_ORDER=10
+CONFIG_ARCH_FORCE_MAX_ORDER=11
 CONFIG_UNMAP_KERNEL_AT_EL0=y
 CONFIG_MITIGATE_SPECTRE_BRANCH_HISTORY=y
 CONFIG_ARM64_SW_TTBR0_PAN=y
@@ -550,6 +548,7 @@ CONFIG_ARM64_TLB_RANGE=y
 #
 CONFIG_AS_HAS_ARMV8_5=y
 CONFIG_ARM64_BTI=y
+CONFIG_ARM64_BTI_KERNEL=y
 CONFIG_CC_HAS_BRANCH_PROT_PAC_RET_BTI=y
 CONFIG_ARM64_E0PD=y
 CONFIG_ARM64_AS_HAS_MTE=y
@@ -885,11 +884,11 @@ CONFIG_HAVE_ARCH_MMAP_RND_BITS=y
 CONFIG_ARCH_MMAP_RND_BITS=18
 CONFIG_HAVE_ARCH_MMAP_RND_COMPAT_BITS=y
 CONFIG_ARCH_MMAP_RND_COMPAT_BITS=11
-CONFIG_HAVE_PAGE_SIZE_4KB=y
-CONFIG_PAGE_SIZE_4KB=y
+CONFIG_HAVE_PAGE_SIZE_16KB=y
+CONFIG_PAGE_SIZE_16KB=y
 CONFIG_PAGE_SIZE_LESS_THAN_64KB=y
 CONFIG_PAGE_SIZE_LESS_THAN_256KB=y
-CONFIG_PAGE_SHIFT=12
+CONFIG_PAGE_SHIFT=14
 CONFIG_ARCH_WANT_DEFAULT_TOPDOWN_MMAP_LAYOUT=y
 CONFIG_HAVE_RELIABLE_STACKTRACE=y
 CONFIG_CLONE_BACKWARDS=y
@@ -936,7 +935,6 @@ CONFIG_FUNCTION_ALIGNMENT_4B=y
 CONFIG_FUNCTION_ALIGNMENT_8B=y
 CONFIG_FUNCTION_ALIGNMENT=8
 CONFIG_CC_HAS_SANE_FUNCTION_ALIGNMENT=y
-CONFIG_HAVE_ARCH_GET_SECUREBOOT=y
 # end of General architecture-dependent options
 
 CONFIG_RT_MUTEXES=y
@@ -954,7 +952,7 @@ CONFIG_ASM_MODVERSIONS=y
 CONFIG_BASIC_MODVERSIONS=y
 CONFIG_MODULE_SRCVERSION_ALL=y
 CONFIG_MODULE_SIG=y
-# CONFIG_MODULE_SIG_FORCE is not set
+CONFIG_MODULE_SIG_FORCE=y
 CONFIG_MODULE_SIG_ALL=y
 # CONFIG_MODULE_SIG_SHA1 is not set
 # CONFIG_MODULE_SIG_SHA256 is not set
@@ -1195,14 +1193,12 @@ CONFIG_KSM=y
 CONFIG_DEFAULT_MMAP_MIN_ADDR=65536
 CONFIG_ARCH_SUPPORTS_MEMORY_FAILURE=y
 # CONFIG_MEMORY_FAILURE is not set
-CONFIG_ARCH_WANTS_THP_SWAP=y
 # CONFIG_PERSISTENT_HUGE_ZERO_FOLIO is not set
 CONFIG_MM_ID=y
 CONFIG_TRANSPARENT_HUGEPAGE=y
 # CONFIG_TRANSPARENT_HUGEPAGE_ALWAYS is not set
 CONFIG_TRANSPARENT_HUGEPAGE_MADVISE=y
 # CONFIG_TRANSPARENT_HUGEPAGE_NEVER is not set
-CONFIG_THP_SWAP=y
 # CONFIG_READ_ONLY_THP_FOR_FS is not set
 # CONFIG_NO_PAGE_MAPCOUNT is not set
 CONFIG_PAGE_MAPCOUNT=y
@@ -1228,7 +1224,6 @@ CONFIG_ARCH_HAS_ZONE_DMA_SET=y
 CONFIG_ZONE_DMA=y
 CONFIG_ZONE_DMA32=y
 CONFIG_ZONE_DEVICE=y
-CONFIG_HMM_MIRROR=y
 CONFIG_GET_FREE_REGION=y
 CONFIG_DEVICE_PRIVATE=y
 CONFIG_ARCH_USES_HIGH_VMA_FLAGS=y
@@ -1388,14 +1383,12 @@ CONFIG_IPV6_SIT=y
 # CONFIG_IPV6_SIT_6RD is not set
 CONFIG_IPV6_NDISC_NODETYPE=y
 CONFIG_IPV6_TUNNEL=y
-CONFIG_IPV6_GRE=m
+# CONFIG_IPV6_GRE is not set
 CONFIG_IPV6_FOU=y
 CONFIG_IPV6_FOU_TUNNEL=y
 CONFIG_IPV6_MULTIPLE_TABLES=y
 # CONFIG_IPV6_SUBTREES is not set
-CONFIG_IPV6_MROUTE=y
-# CONFIG_IPV6_MROUTE_MULTIPLE_TABLES is not set
-# CONFIG_IPV6_PIMSM_V2 is not set
+# CONFIG_IPV6_MROUTE is not set
 CONFIG_IPV6_SEG6_LWTUNNEL=y
 # CONFIG_IPV6_SEG6_HMAC is not set
 CONFIG_IPV6_SEG6_BPF=y
@@ -1405,7 +1398,7 @@ CONFIG_NETLABEL=y
 # CONFIG_MPTCP is not set
 CONFIG_NETWORK_SECMARK=y
 CONFIG_NET_PTP_CLASSIFY=y
-# CONFIG_NETWORK_PHY_TIMESTAMPING is not set
+CONFIG_NETWORK_PHY_TIMESTAMPING=y
 CONFIG_NETFILTER=y
 CONFIG_NETFILTER_ADVANCED=y
 CONFIG_BRIDGE_NETFILTER=y
@@ -1468,7 +1461,7 @@ CONFIG_NF_TABLES_NETDEV=y
 CONFIG_NFT_NUMGEN=y
 CONFIG_NFT_CT=y
 # CONFIG_NFT_EXTHDR_DCCP is not set
-CONFIG_NFT_CONNLIMIT=y
+# CONFIG_NFT_CONNLIMIT is not set
 CONFIG_NFT_LOG=y
 CONFIG_NFT_LIMIT=y
 CONFIG_NFT_MASQ=y
@@ -1859,7 +1852,7 @@ CONFIG_NET_ACT_TUNNEL_KEY=y
 # CONFIG_NET_IFE_SKBMARK is not set
 # CONFIG_NET_IFE_SKBPRIO is not set
 # CONFIG_NET_IFE_SKBTCINDEX is not set
-CONFIG_NET_TC_SKB_EXT=y
+# CONFIG_NET_TC_SKB_EXT is not set
 CONFIG_NET_SCH_FIFO=y
 CONFIG_DCB=y
 CONFIG_DNS_RESOLVER=y
@@ -2050,7 +2043,7 @@ CONFIG_PCIE_DW=y
 CONFIG_PCIE_DW_HOST=y
 # CONFIG_PCIE_AL is not set
 # CONFIG_PCIE_AMD_MDB is not set
-CONFIG_PCI_MESON=y
+# CONFIG_PCI_MESON is not set
 # CONFIG_PCI_IMX6_HOST is not set
 # CONFIG_PCI_IMX6_EP is not set
 CONFIG_PCI_LAYERSCAPE=y
@@ -2237,6 +2230,7 @@ CONFIG_DMIID=y
 CONFIG_DMI_SYSFS=y
 CONFIG_ISCSI_IBFT=y
 CONFIG_RASPBERRYPI_FIRMWARE=y
+CONFIG_FIRMWARE_RP1=y
 CONFIG_FW_CFG_SYSFS=y
 # CONFIG_FW_CFG_SYSFS_CMDLINE is not set
 CONFIG_SYSFB=y
@@ -2277,7 +2271,6 @@ CONFIG_UEFI_CPER_ARM=y
 CONFIG_IMX_SCMI_CPU_DRV=y
 CONFIG_IMX_SCMI_LMM_DRV=y
 CONFIG_IMX_SCMI_MISC_DRV=y
-CONFIG_MESON_SM=y
 CONFIG_ARM_PSCI_FW=y
 # CONFIG_ARM_PSCI_CHECKER is not set
 
@@ -2429,6 +2422,7 @@ CONFIG_OF_RESERVED_MEM=y
 CONFIG_OF_RESOLVE=y
 CONFIG_OF_OVERLAY=y
 CONFIG_OF_NUMA=y
+# CONFIG_OF_CONFIGFS is not set
 # CONFIG_PARPORT is not set
 CONFIG_PNP=y
 CONFIG_PNP_DEBUG_MESSAGES=y
@@ -2461,7 +2455,7 @@ CONFIG_BLK_DEV_UBLK=m
 #
 CONFIG_NVME_AUTH=m
 CONFIG_NVME_CORE=y
-CONFIG_BLK_DEV_NVME=m
+CONFIG_BLK_DEV_NVME=y
 CONFIG_NVME_MULTIPATH=y
 # CONFIG_NVME_VERBOSE_ERRORS is not set
 CONFIG_NVME_HWMON=y
@@ -2487,6 +2481,9 @@ CONFIG_NVME_TARGET_AUTH=y
 #
 # Misc devices
 #
+CONFIG_BCM2835_SMI=m
+CONFIG_RP1_PIO=y
+CONFIG_WS2812_PIO_RP1=y
 # CONFIG_AD525X_DPOT is not set
 # CONFIG_DUMMY_IRQ is not set
 # CONFIG_PHANTOM is not set
@@ -2496,7 +2493,7 @@ CONFIG_NVME_TARGET_AUTH=y
 # CONFIG_ICS932S401 is not set
 CONFIG_ENCLOSURE_SERVICES=y
 # CONFIG_HI6421V600_IRQ is not set
-CONFIG_HP_ILO=m
+CONFIG_HP_ILO=y
 # CONFIG_QCOM_COINCELL is not set
 # CONFIG_QCOM_FASTRPC is not set
 # CONFIG_APDS9802ALS is not set
@@ -2548,7 +2545,7 @@ CONFIG_EEPROM_93CX6=y
 # CONFIG_GP_PCI1XXXX is not set
 # CONFIG_KEBA_CP500 is not set
 # CONFIG_AMD_SBRMI_I2C is not set
-CONFIG_MISC_RP1=y
+# CONFIG_MISC_RP1 is not set
 # end of Misc devices
 
 #
@@ -2598,7 +2595,7 @@ CONFIG_ISCSI_BOOT_SYSFS=y
 # CONFIG_SCSI_BNX2X_FCOE is not set
 # CONFIG_BE2ISCSI is not set
 # CONFIG_BLK_DEV_3W_XXXX_RAID is not set
-CONFIG_SCSI_HPSA=m
+CONFIG_SCSI_HPSA=y
 # CONFIG_SCSI_3W_9XXX is not set
 # CONFIG_SCSI_3W_SAS is not set
 # CONFIG_SCSI_ACARD is not set
@@ -2616,13 +2613,13 @@ CONFIG_SCSI_HISI_SAS_PCI=m
 # CONFIG_SCSI_ESAS2R is not set
 # CONFIG_MEGARAID_NEWGEN is not set
 # CONFIG_MEGARAID_LEGACY is not set
-CONFIG_MEGARAID_SAS=m
-CONFIG_SCSI_MPT3SAS=m
+CONFIG_MEGARAID_SAS=y
+CONFIG_SCSI_MPT3SAS=y
 CONFIG_SCSI_MPT2SAS_MAX_SGE=128
 CONFIG_SCSI_MPT3SAS_MAX_SGE=128
 # CONFIG_SCSI_MPT2SAS is not set
 CONFIG_SCSI_MPI3MR=m
-CONFIG_SCSI_SMARTPQI=m
+CONFIG_SCSI_SMARTPQI=y
 # CONFIG_SCSI_HPTIOP is not set
 # CONFIG_SCSI_BUSLOGIC is not set
 # CONFIG_SCSI_MYRB is not set
@@ -2678,7 +2675,7 @@ CONFIG_SATA_PMP=y
 #
 # Controllers with non-SFF native interface
 #
-CONFIG_SATA_AHCI=m
+CONFIG_SATA_AHCI=y
 CONFIG_SATA_MOBILE_LPM_POLICY=0
 CONFIG_SATA_AHCI_PLATFORM=y
 # CONFIG_AHCI_BRCM is not set
@@ -2725,7 +2722,7 @@ CONFIG_SATA_RCAR=y
 # PATA SFF controllers with BMDMA
 #
 # CONFIG_PATA_ALI is not set
-CONFIG_PATA_AMD=m
+CONFIG_PATA_AMD=y
 # CONFIG_PATA_ARTOP is not set
 # CONFIG_PATA_ATIIXP is not set
 # CONFIG_PATA_ATP867X is not set
@@ -2740,17 +2737,17 @@ CONFIG_PATA_AMD=m
 # CONFIG_PATA_IT8213 is not set
 # CONFIG_PATA_IT821X is not set
 # CONFIG_PATA_JMICRON is not set
-CONFIG_PATA_MARVELL=m
+CONFIG_PATA_MARVELL=y
 # CONFIG_PATA_NETCELL is not set
 # CONFIG_PATA_NINJA32 is not set
 # CONFIG_PATA_NS87415 is not set
-CONFIG_PATA_OLDPIIX=m
+CONFIG_PATA_OLDPIIX=y
 # CONFIG_PATA_OPTIDMA is not set
 # CONFIG_PATA_PDC2027X is not set
 # CONFIG_PATA_PDC_OLD is not set
 # CONFIG_PATA_RADISYS is not set
 # CONFIG_PATA_RDC is not set
-CONFIG_PATA_SCH=m
+CONFIG_PATA_SCH=y
 # CONFIG_PATA_SERVERWORKS is not set
 # CONFIG_PATA_SIL680 is not set
 # CONFIG_PATA_SIS is not set
@@ -2787,7 +2784,7 @@ CONFIG_MD_RAID0=y
 CONFIG_MD_RAID1=y
 CONFIG_MD_RAID10=y
 CONFIG_MD_RAID456=m
-CONFIG_BCACHE=m
+CONFIG_BCACHE=y
 # CONFIG_BCACHE_DEBUG is not set
 # CONFIG_BCACHE_ASYNC_REGISTRATION is not set
 CONFIG_BLK_DEV_DM_BUILTIN=y
@@ -2824,8 +2821,8 @@ CONFIG_DM_MULTIPATH_ST=m
 # CONFIG_DM_VERITY is not set
 # CONFIG_DM_SWITCH is not set
 # CONFIG_DM_LOG_WRITES is not set
-CONFIG_DM_INTEGRITY=m
-CONFIG_DM_AUDIT=y
+# CONFIG_DM_INTEGRITY is not set
+# CONFIG_DM_AUDIT is not set
 # CONFIG_DM_VDO is not set
 CONFIG_TARGET_CORE=m
 # CONFIG_TCM_IBLOCK is not set
@@ -2914,7 +2911,7 @@ CONFIG_NET_VRF=m
 # end of Distributed Switch Architecture drivers
 
 CONFIG_ETHERNET=y
-CONFIG_MDIO=m
+CONFIG_MDIO=y
 CONFIG_NET_VENDOR_3COM=y
 # CONFIG_VORTEX is not set
 # CONFIG_TYPHOON is not set
@@ -2956,16 +2953,16 @@ CONFIG_ALX=m
 CONFIG_NET_VENDOR_BROADCOM=y
 # CONFIG_B44 is not set
 CONFIG_BCMGENET=y
-CONFIG_BNX2=m
+CONFIG_BNX2=y
 # CONFIG_CNIC is not set
-CONFIG_TIGON3=m
+CONFIG_TIGON3=y
 CONFIG_TIGON3_HWMON=y
-CONFIG_BNX2X=m
+CONFIG_BNX2X=y
 CONFIG_BNX2X_SRIOV=y
 CONFIG_BGMAC=y
 CONFIG_BGMAC_PLATFORM=y
 # CONFIG_SYSTEMPORT is not set
-CONFIG_BNXT=m
+CONFIG_BNXT=y
 CONFIG_BNXT_SRIOV=y
 CONFIG_BNXT_FLOWER_OFFLOAD=y
 # CONFIG_BNXT_DCB is not set
@@ -2981,7 +2978,7 @@ CONFIG_NET_VENDOR_CAVIUM=y
 # CONFIG_THUNDER_NIC_VF is not set
 # CONFIG_THUNDER_NIC_BGX is not set
 # CONFIG_THUNDER_NIC_RGX is not set
-CONFIG_CAVIUM_PTP=m
+CONFIG_CAVIUM_PTP=y
 # CONFIG_LIQUIDIO is not set
 # CONFIG_LIQUIDIO_VF is not set
 CONFIG_NET_VENDOR_CHELSIO=y
@@ -2994,7 +2991,7 @@ CONFIG_CHELSIO_T4VF=m
 CONFIG_CHELSIO_INLINE_CRYPTO=y
 CONFIG_CHELSIO_IPSEC_INLINE=m
 CONFIG_NET_VENDOR_CISCO=y
-CONFIG_ENIC=m
+CONFIG_ENIC=y
 CONFIG_NET_VENDOR_CORTINA=y
 # CONFIG_GEMINI_ETHERNET is not set
 CONFIG_NET_VENDOR_DAVICOM=y
@@ -3022,6 +3019,7 @@ CONFIG_NET_VENDOR_FREESCALE=y
 # CONFIG_FSL_PQ_MDIO is not set
 # CONFIG_FSL_XGMAC_MDIO is not set
 # CONFIG_GIANFAR is not set
+# CONFIG_FSL_DPAA2_SWITCH is not set
 # CONFIG_FSL_ENETC is not set
 # CONFIG_NXP_ENETC4 is not set
 # CONFIG_FSL_ENETC_VF is not set
@@ -3052,34 +3050,32 @@ CONFIG_NET_VENDOR_HUAWEI=y
 # CONFIG_HINIC3 is not set
 CONFIG_NET_VENDOR_I825XX=y
 CONFIG_NET_VENDOR_INTEL=y
-CONFIG_LIBETH=m
-CONFIG_LIBETH_XDP=m
-CONFIG_LIBIE=m
-CONFIG_LIBIE_ADMINQ=m
-CONFIG_LIBIE_FWLOG=m
-CONFIG_E100=m
-CONFIG_E1000=m
-CONFIG_E1000E=m
-CONFIG_IGB=m
+CONFIG_LIBETH=y
+CONFIG_LIBIE=y
+CONFIG_LIBIE_ADMINQ=y
+CONFIG_LIBIE_FWLOG=y
+CONFIG_E100=y
+CONFIG_E1000=y
+CONFIG_E1000E=y
+CONFIG_IGB=y
 CONFIG_IGB_HWMON=y
-CONFIG_IGBVF=m
-CONFIG_IXGBE=m
+CONFIG_IGBVF=y
+CONFIG_IXGBE=y
 CONFIG_IXGBE_HWMON=y
 # CONFIG_IXGBE_DCB is not set
 CONFIG_IXGBE_IPSEC=y
-CONFIG_IXGBEVF=m
+CONFIG_IXGBEVF=y
 CONFIG_IXGBEVF_IPSEC=y
-CONFIG_I40E=m
+CONFIG_I40E=y
 # CONFIG_I40E_DCB is not set
-CONFIG_IAVF=m
-CONFIG_I40EVF=m
-CONFIG_ICE=m
+CONFIG_IAVF=y
+CONFIG_I40EVF=y
+CONFIG_ICE=y
 CONFIG_ICE_HWMON=y
 CONFIG_ICE_SWITCHDEV=y
 # CONFIG_FM10K is not set
-CONFIG_IGC=m
-CONFIG_IDPF=m
-# CONFIG_IDPF_SINGLEQ is not set
+CONFIG_IGC=y
+# CONFIG_IDPF is not set
 # CONFIG_JME is not set
 CONFIG_NET_VENDOR_ADI=y
 # CONFIG_ADIN1110 is not set
@@ -3092,7 +3088,7 @@ CONFIG_MVPP2=y
 # CONFIG_MVPP2_PTP is not set
 # CONFIG_PXA168_ETH is not set
 # CONFIG_SKGE is not set
-CONFIG_SKY2=m
+CONFIG_SKY2=y
 # CONFIG_SKY2_DEBUG is not set
 # CONFIG_OCTEONTX2_AF is not set
 # CONFIG_OCTEONTX2_PF is not set
@@ -3114,16 +3110,13 @@ CONFIG_MLX5_EN_RXNFC=y
 CONFIG_MLX5_MPFS=y
 CONFIG_MLX5_ESWITCH=y
 CONFIG_MLX5_BRIDGE=y
-CONFIG_MLX5_CLS_ACT=y
-CONFIG_MLX5_TC_SAMPLE=y
 CONFIG_MLX5_CORE_EN_DCB=y
 CONFIG_MLX5_CORE_IPOIB=y
 # CONFIG_MLX5_MACSEC is not set
 # CONFIG_MLX5_EN_IPSEC is not set
 CONFIG_MLX5_SW_STEERING=y
 CONFIG_MLX5_HW_STEERING=y
-CONFIG_MLX5_SF=y
-CONFIG_MLX5_SF_MANAGER=y
+# CONFIG_MLX5_SF is not set
 # CONFIG_MLX5_DPLL is not set
 CONFIG_MLXSW_CORE=m
 CONFIG_MLXSW_CORE_HWMON=y
@@ -3177,15 +3170,15 @@ CONFIG_NET_VENDOR_PENSANDO=y
 # CONFIG_IONIC is not set
 CONFIG_NET_VENDOR_QLOGIC=y
 # CONFIG_QLA3XXX is not set
-CONFIG_QLCNIC=m
+CONFIG_QLCNIC=y
 CONFIG_QLCNIC_SRIOV=y
 CONFIG_QLCNIC_DCB=y
 CONFIG_QLCNIC_HWMON=y
 # CONFIG_NETXEN_NIC is not set
-CONFIG_QED=m
+CONFIG_QED=y
 CONFIG_QED_LL2=y
 CONFIG_QED_SRIOV=y
-CONFIG_QEDE=m
+CONFIG_QEDE=y
 CONFIG_QED_FCOE=y
 CONFIG_NET_VENDOR_BROCADE=y
 # CONFIG_BNA is not set
@@ -3198,12 +3191,12 @@ CONFIG_NET_VENDOR_RDC=y
 # CONFIG_R6040 is not set
 CONFIG_NET_VENDOR_REALTEK=y
 # CONFIG_8139CP is not set
-CONFIG_8139TOO=m
+CONFIG_8139TOO=y
 CONFIG_8139TOO_PIO=y
 # CONFIG_8139TOO_TUNE_TWISTER is not set
 # CONFIG_8139TOO_8129 is not set
 # CONFIG_8139_OLD_RX_RESET is not set
-CONFIG_R8169=m
+CONFIG_R8169=y
 # CONFIG_RTASE is not set
 CONFIG_NET_VENDOR_RENESAS=y
 # CONFIG_SH_ETH is not set
@@ -3240,24 +3233,24 @@ CONFIG_SMSC911X=y
 CONFIG_NET_VENDOR_SOCIONEXT=y
 # CONFIG_SNI_NETSEC is not set
 CONFIG_NET_VENDOR_STMICRO=y
-CONFIG_STMMAC_ETH=m
+CONFIG_STMMAC_ETH=y
 # CONFIG_STMMAC_SELFTESTS is not set
-CONFIG_STMMAC_PLATFORM=m
-CONFIG_DWMAC_DWC_QOS_ETH=m
-CONFIG_DWMAC_GENERIC=m
-CONFIG_DWMAC_IPQ806X=m
+CONFIG_STMMAC_PLATFORM=y
+CONFIG_DWMAC_DWC_QOS_ETH=y
+CONFIG_DWMAC_GENERIC=y
+CONFIG_DWMAC_IPQ806X=y
 # CONFIG_DWMAC_MEDIATEK is not set
-CONFIG_DWMAC_MESON=m
-CONFIG_DWMAC_QCOM_ETHQOS=m
-CONFIG_DWMAC_RENESAS_GBETH=m
-CONFIG_DWMAC_ROCKCHIP=m
-CONFIG_DWMAC_SUNXI=m
-CONFIG_DWMAC_SUN8I=m
-CONFIG_DWMAC_SUN55I=m
-CONFIG_DWMAC_IMX8=m
+CONFIG_DWMAC_MESON=y
+CONFIG_DWMAC_QCOM_ETHQOS=y
+CONFIG_DWMAC_RENESAS_GBETH=y
+CONFIG_DWMAC_ROCKCHIP=y
+CONFIG_DWMAC_SUNXI=y
+CONFIG_DWMAC_SUN8I=y
+CONFIG_DWMAC_SUN55I=y
+CONFIG_DWMAC_IMX8=y
 # CONFIG_DWMAC_INTEL_PLAT is not set
-CONFIG_DWMAC_TEGRA=m
-CONFIG_STMMAC_PCI=m
+CONFIG_DWMAC_TEGRA=y
+CONFIG_STMMAC_PCI=y
 CONFIG_NET_VENDOR_SUN=y
 # CONFIG_HAPPYMEAL is not set
 # CONFIG_SUNGEM is not set
@@ -3319,6 +3312,7 @@ CONFIG_BCM84881_PHY=y
 CONFIG_BCM87XX_PHY=y
 # CONFIG_BCM_CYGNUS_PHY is not set
 CONFIG_BCM_NET_PHYLIB=y
+CONFIG_BCM_NET_PHYPTP=y
 CONFIG_CICADA_PHY=y
 CONFIG_CORTINA_PHY=y
 CONFIG_DAVICOM_PHY=y
@@ -3338,6 +3332,7 @@ CONFIG_MICREL_PHY=y
 # CONFIG_MICROCHIP_T1S_PHY is not set
 CONFIG_MICROCHIP_PHY=y
 CONFIG_MICROCHIP_T1_PHY=y
+CONFIG_MICROCHIP_PHY_RDS_PTP=y
 CONFIG_MICROSEMI_PHY=m
 # CONFIG_MOTORCOMM_PHY is not set
 CONFIG_NATIONAL_PHY=y
@@ -3361,7 +3356,7 @@ CONFIG_TERANETICS_PHY=y
 CONFIG_DP83822_PHY=y
 CONFIG_DP83TC811_PHY=y
 CONFIG_DP83848_PHY=y
-CONFIG_DP83867_PHY=m
+CONFIG_DP83867_PHY=y
 CONFIG_DP83869_PHY=y
 # CONFIG_DP83TD510_PHY is not set
 # CONFIG_DP83TG720_PHY is not set
@@ -3400,8 +3395,8 @@ CONFIG_MDIO_BUS_MUX_MMIOREG=y
 #
 # PCS device drivers
 #
-CONFIG_PCS_XPCS=m
-CONFIG_PCS_RZN1_MIIC=m
+CONFIG_PCS_XPCS=y
+CONFIG_PCS_RZN1_MIIC=y
 # end of PCS device drivers
 
 CONFIG_PPP=y
@@ -3423,7 +3418,7 @@ CONFIG_USB_NET_DRIVERS=y
 # CONFIG_USB_KAWETH is not set
 CONFIG_USB_PEGASUS=y
 CONFIG_USB_RTL8150=y
-CONFIG_USB_RTL8152=m
+CONFIG_USB_RTL8152=y
 CONFIG_USB_LAN78XX=y
 CONFIG_USB_USBNET=m
 CONFIG_USB_NET_AX8817X=m
@@ -3456,7 +3451,7 @@ CONFIG_USB_NET_CX82310_ETH=m
 CONFIG_USB_NET_KALMIA=m
 CONFIG_USB_NET_QMI_WWAN=m
 CONFIG_USB_NET_INT51X1=m
-CONFIG_USB_IPHETH=m
+# CONFIG_USB_IPHETH is not set
 CONFIG_USB_SIERRA_NET=m
 CONFIG_USB_VL600=m
 CONFIG_USB_NET_CH9200=m
@@ -3472,7 +3467,7 @@ CONFIG_USB_RTL8153_ECM=m
 # end of Wireless WAN
 
 CONFIG_XEN_NETDEV_FRONTEND=m
-CONFIG_VMXNET3=m
+CONFIG_VMXNET3=y
 # CONFIG_FUJITSU_ES is not set
 CONFIG_USB4_NET=m
 CONFIG_HYPERV_NET=m
@@ -3497,7 +3492,7 @@ CONFIG_INPUT_MOUSEDEV=y
 # CONFIG_INPUT_MOUSEDEV_PSAUX is not set
 CONFIG_INPUT_MOUSEDEV_SCREEN_X=1024
 CONFIG_INPUT_MOUSEDEV_SCREEN_Y=768
-CONFIG_INPUT_JOYDEV=m
+# CONFIG_INPUT_JOYDEV is not set
 CONFIG_INPUT_EVDEV=y
 
 #
@@ -3631,6 +3626,7 @@ CONFIG_INPUT_TOUCHSCREEN=y
 # CONFIG_TOUCHSCREEN_HYNITRON_CST816X is not set
 # CONFIG_TOUCHSCREEN_ILI210X is not set
 # CONFIG_TOUCHSCREEN_ILITEK is not set
+# CONFIG_TOUCHSCREEN_ILITEK_V3 is not set
 # CONFIG_TOUCHSCREEN_IPROC is not set
 # CONFIG_TOUCHSCREEN_S6SY761 is not set
 # CONFIG_TOUCHSCREEN_GUNZE is not set
@@ -3745,6 +3741,8 @@ CONFIG_HYPERV_KEYBOARD=m
 #
 # Character devices
 #
+# CONFIG_BRCM_CHAR_DRIVERS is not set
+CONFIG_BCM2835_SMI_DEV=m
 CONFIG_TTY=y
 CONFIG_VT=y
 CONFIG_CONSOLE_TRANSLATIONS=y
@@ -3837,6 +3835,7 @@ CONFIG_SERIAL_XILINX_PS_UART_CONSOLE=y
 CONFIG_SERIAL_MVEBU_UART=y
 CONFIG_SERIAL_MVEBU_CONSOLE=y
 # CONFIG_SERIAL_OWL is not set
+# CONFIG_SERIAL_RPI_FW is not set
 # end of Serial drivers
 
 CONFIG_SERIAL_MCTRL_GPIO=y
@@ -3912,6 +3911,7 @@ CONFIG_TCG_CRB=y
 # CONFIG_TCG_TIS_ST33ZP24_SPI is not set
 # CONFIG_XILLYBUS is not set
 # CONFIG_XILLYUSB is not set
+CONFIG_RASPBERRYPI_GPIOMEM=y
 # end of Character devices
 
 #
@@ -3921,7 +3921,7 @@ CONFIG_I2C=y
 CONFIG_ACPI_I2C_OPREGION=y
 CONFIG_I2C_BOARDINFO=y
 CONFIG_I2C_CHARDEV=y
-CONFIG_I2C_MUX=m
+CONFIG_I2C_MUX=y
 
 #
 # Multiplexer I2C Chip support
@@ -3940,7 +3940,7 @@ CONFIG_I2C_MUX=m
 
 CONFIG_I2C_HELPER_AUTO=y
 CONFIG_I2C_SMBUS=y
-CONFIG_I2C_ALGOBIT=m
+CONFIG_I2C_ALGOBIT=y
 
 #
 # I2C Hardware Bus support
@@ -3950,6 +3950,8 @@ CONFIG_I2C_ALGOBIT=m
 # PC SMBus host controller drivers
 #
 CONFIG_I2C_CCGX_UCSI=y
+CONFIG_I2C_BCM2708=y
+CONFIG_I2C_BCM2708_BAUDRATE=100000
 # CONFIG_I2C_ALI1535 is not set
 # CONFIG_I2C_ALI1563 is not set
 # CONFIG_I2C_ALI15X3 is not set
@@ -3957,7 +3959,7 @@ CONFIG_I2C_CCGX_UCSI=y
 # CONFIG_I2C_AMD8111 is not set
 # CONFIG_I2C_AMD_MP2 is not set
 # CONFIG_I2C_HIX5HD2 is not set
-CONFIG_I2C_I801=m
+CONFIG_I2C_I801=y
 # CONFIG_I2C_ISCH is not set
 # CONFIG_I2C_PIIX4 is not set
 # CONFIG_I2C_NFORCE2 is not set
@@ -4063,7 +4065,7 @@ CONFIG_SPI_MEM=y
 CONFIG_SPI_BCM2835=y
 CONFIG_SPI_BCM2835AUX=y
 CONFIG_SPI_BCM_QSPI=y
-# CONFIG_SPI_BITBANG is not set
+CONFIG_SPI_BITBANG=y
 # CONFIG_SPI_CADENCE is not set
 # CONFIG_SPI_CADENCE_QUADSPI is not set
 # CONFIG_SPI_CADENCE_XSPI is not set
@@ -4074,7 +4076,7 @@ CONFIG_SPI_BCM_QSPI=y
 # CONFIG_SPI_HISI_KUNPENG is not set
 # CONFIG_SPI_HISI_SFC_V3XX is not set
 # CONFIG_SPI_NXP_FLEXSPI is not set
-# CONFIG_SPI_GPIO is not set
+CONFIG_SPI_GPIO=y
 # CONFIG_SPI_IMX is not set
 # CONFIG_SPI_FSL_SPI is not set
 # CONFIG_SPI_FSL_DSPI is not set
@@ -4090,6 +4092,7 @@ CONFIG_SPI_ORION=y
 CONFIG_SPI_PL022=y
 CONFIG_SPI_ROCKCHIP=y
 # CONFIG_SPI_ROCKCHIP_SFC is not set
+CONFIG_SPI_RP2040_GPIO_BRIDGE=y
 # CONFIG_SPI_RSPI is not set
 # CONFIG_SPI_RZV2H_RSPI is not set
 # CONFIG_SPI_RZV2M_CSI is not set
@@ -4153,10 +4156,8 @@ CONFIG_PPS=y
 CONFIG_PTP_1588_CLOCK=y
 CONFIG_PTP_1588_CLOCK_OPTIONAL=y
 CONFIG_PTP_1588_CLOCK_DTE=y
-
-#
-# Enable PHYLIB and NETWORK_PHY_TIMESTAMPING to see the additional clocks.
-#
+# CONFIG_DP83640_PHY is not set
+# CONFIG_PTP_1588_CLOCK_INES is not set
 CONFIG_PTP_1588_CLOCK_KVM=y
 CONFIG_PTP_1588_CLOCK_VMCLOCK=y
 # CONFIG_PTP_1588_CLOCK_IDT82P33 is not set
@@ -4188,7 +4189,9 @@ CONFIG_PINCTRL_AXP209=y
 CONFIG_PINCTRL_BM1880=y
 # CONFIG_PINCTRL_CY8C95X0 is not set
 CONFIG_PINCTRL_MAX77620=y
-# CONFIG_PINCTRL_MCP23S08 is not set
+CONFIG_PINCTRL_MCP23S08_I2C=y
+CONFIG_PINCTRL_MCP23S08_SPI=y
+CONFIG_PINCTRL_MCP23S08=y
 # CONFIG_PINCTRL_MICROCHIP_SGPIO is not set
 # CONFIG_PINCTRL_OCELOT is not set
 CONFIG_PINCTRL_RK805=y
@@ -4384,6 +4387,7 @@ CONFIG_GPIO_GENERIC=y
 # CONFIG_GPIO_AMDPT is not set
 CONFIG_GPIO_RASPBERRYPI_EXP=y
 CONFIG_GPIO_BCM_XGS_IPROC=y
+CONFIG_GPIO_BCM_VIRT=y
 CONFIG_GPIO_BRCMSTB=y
 # CONFIG_GPIO_CADENCE is not set
 # CONFIG_GPIO_DAVINCI is not set
@@ -4402,6 +4406,7 @@ CONFIG_GPIO_MVEBU=y
 CONFIG_GPIO_MXC=y
 CONFIG_GPIO_PL061=y
 # CONFIG_GPIO_POLARFIRE_SOC is not set
+# CONFIG_GPIO_PWM is not set
 CONFIG_GPIO_RCAR=y
 CONFIG_GPIO_ROCKCHIP=y
 CONFIG_GPIO_RTD=y
@@ -4440,6 +4445,7 @@ CONFIG_GPIO_PCA953X_IRQ=y
 #
 # MFD GPIO expanders
 #
+# CONFIG_GPIO_FSM is not set
 CONFIG_GPIO_MAX77620=y
 # end of MFD GPIO expanders
 
@@ -4509,6 +4515,7 @@ CONFIG_POWER_RESET_SYSCON=y
 CONFIG_POWER_SUPPLY=y
 # CONFIG_POWER_SUPPLY_DEBUG is not set
 CONFIG_POWER_SUPPLY_HWMON=y
+# CONFIG_RPI_POE_POWER is not set
 # CONFIG_GENERIC_ADC_BATTERY is not set
 # CONFIG_IP5XXX_POWER is not set
 # CONFIG_TEST_POWER is not set
@@ -4610,7 +4617,7 @@ CONFIG_SENSORS_I5K_AMB=m
 # CONFIG_SENSORS_GL520SM is not set
 # CONFIG_SENSORS_G760A is not set
 # CONFIG_SENSORS_G762 is not set
-CONFIG_SENSORS_GPIO_FAN=y
+# CONFIG_SENSORS_GPIO_FAN is not set
 # CONFIG_SENSORS_HIH6130 is not set
 # CONFIG_SENSORS_HS3001 is not set
 # CONFIG_SENSORS_HTU31 is not set
@@ -4744,6 +4751,7 @@ CONFIG_SENSORS_SPD5118_DETECT=y
 # CONFIG_SENSORS_W83627HF is not set
 # CONFIG_SENSORS_W83627EHF is not set
 # CONFIG_SENSORS_XGENE is not set
+# CONFIG_SENSORS_RP1_ADC is not set
 
 #
 # ACPI drivers
@@ -4794,7 +4802,7 @@ CONFIG_AMLOGIC_THERMAL=y
 # Broadcom thermal drivers
 #
 CONFIG_BCM2711_THERMAL=y
-# CONFIG_BCM2835_THERMAL is not set
+CONFIG_BCM2835_THERMAL=y
 # CONFIG_BRCMSTB_THERMAL is not set
 CONFIG_BCM_NS_THERMAL=y
 CONFIG_BCM_SR_THERMAL=y
@@ -4973,6 +4981,7 @@ CONFIG_MFD_MAX77620=y
 # CONFIG_MFD_QCOM_RPM is not set
 CONFIG_MFD_SPMI_PMIC=y
 # CONFIG_MFD_SY7636A is not set
+CONFIG_MFD_RASPBERRYPI_POE_HAT=y
 # CONFIG_MFD_RDC321X is not set
 # CONFIG_MFD_RT4831 is not set
 # CONFIG_MFD_RT5033 is not set
@@ -4984,6 +4993,7 @@ CONFIG_MFD_RK8XX_SPI=y
 # CONFIG_MFD_RN5T618 is not set
 # CONFIG_MFD_SEC_I2C is not set
 # CONFIG_MFD_SI476X_CORE is not set
+CONFIG_MFD_SIMPLE_MFD_I2C=y
 # CONFIG_MFD_SL28CPLD is not set
 # CONFIG_MFD_SM501 is not set
 # CONFIG_MFD_SKY81452 is not set
@@ -5038,6 +5048,7 @@ CONFIG_MFD_SYSCON=y
 # CONFIG_MFD_QCOM_PM8008 is not set
 # CONFIG_MFD_INTEL_M10_BMC_SPI is not set
 # CONFIG_MFD_LS2K_BMC_CORE is not set
+CONFIG_MFD_RP1=y
 # CONFIG_MFD_RSMU_I2C is not set
 # CONFIG_MFD_RSMU_SPI is not set
 # CONFIG_MFD_MAX7360 is not set
@@ -5108,6 +5119,7 @@ CONFIG_REGULATOR_QCOM_SPMI=y
 # CONFIG_REGULATOR_RAA215300 is not set
 # CONFIG_REGULATOR_RASPBERRYPI_TOUCHSCREEN_ATTINY is not set
 # CONFIG_REGULATOR_RASPBERRYPI_TOUCHSCREEN_V2 is not set
+# CONFIG_REGULATOR_WAVESHARE_TOUCHSCREEN is not set
 CONFIG_REGULATOR_RK808=y
 # CONFIG_REGULATOR_RT4801 is not set
 # CONFIG_REGULATOR_RT4803 is not set
@@ -5242,6 +5254,7 @@ CONFIG_V4L_MEM2MEM_DRIVERS=y
 #
 # Atmel media platform drivers
 #
+# CONFIG_VIDEO_BCM2835_UNICAM_LEGACY is not set
 # CONFIG_VIDEO_BCM2835_UNICAM is not set
 
 #
@@ -5304,8 +5317,10 @@ CONFIG_V4L_MEM2MEM_DRIVERS=y
 #
 # Raspberry Pi media platform drivers
 #
+# CONFIG_VIDEO_RPI_HEVC_DEC is not set
 # CONFIG_VIDEO_RASPBERRYPI_PISP_BE is not set
 # CONFIG_VIDEO_RP1_CFE is not set
+# CONFIG_VIDEO_RP1_CFE_DOWNSTREAM is not set
 
 #
 # Renesas media platform drivers
@@ -5381,8 +5396,10 @@ CONFIG_VIDEOBUF2_VMALLOC=m
 # end of Camera ISPs
 
 CONFIG_VIDEO_CAMERA_LENS=y
+# CONFIG_VIDEO_AD5398 is not set
 # CONFIG_VIDEO_AD5820 is not set
 # CONFIG_VIDEO_AK7375 is not set
+# CONFIG_VIDEO_BU64754 is not set
 # CONFIG_VIDEO_DW9714 is not set
 # CONFIG_VIDEO_DW9719 is not set
 # CONFIG_VIDEO_DW9768 is not set
@@ -5451,6 +5468,7 @@ CONFIG_VIDEO_CAMERA_LENS=y
 # CONFIG_VIDEO_TW9903 is not set
 # CONFIG_VIDEO_TW9906 is not set
 # CONFIG_VIDEO_TW9910 is not set
+# CONFIG_VIDEO_IRS1125 is not set
 # CONFIG_VIDEO_VPX3220 is not set
 
 #
@@ -5524,7 +5542,6 @@ CONFIG_APERTURE_HELPERS=y
 CONFIG_SCREEN_INFO=y
 CONFIG_VIDEO=y
 # CONFIG_AUXDISPLAY is not set
-CONFIG_GPU_BUDDY=y
 CONFIG_TEGRA_HOST1X_CONTEXT_BUS=y
 CONFIG_TEGRA_HOST1X=m
 CONFIG_TEGRA_HOST1X_FIREWALL=y
@@ -5558,7 +5575,7 @@ CONFIG_DRM_CLIENT_DEFAULT_FBDEV=y
 CONFIG_DRM_CLIENT_DEFAULT="fbdev"
 # end of Supported DRM clients
 
-# CONFIG_DRM_LOAD_EDID_FIRMWARE is not set
+CONFIG_DRM_LOAD_EDID_FIRMWARE=y
 CONFIG_DRM_DISPLAY_DP_AUX_BUS=y
 CONFIG_DRM_DISPLAY_HELPER=y
 CONFIG_DRM_BRIDGE_CONNECTOR=y
@@ -5574,14 +5591,13 @@ CONFIG_DRM_DISPLAY_HDMI_HELPER=y
 CONFIG_DRM_DISPLAY_HDMI_STATE_HELPER=y
 CONFIG_DRM_TTM=m
 CONFIG_DRM_EXEC=m
-CONFIG_DRM_GPUVM=m
 CONFIG_DRM_BUDDY=m
 CONFIG_DRM_VRAM_HELPER=m
 CONFIG_DRM_TTM_HELPER=m
 CONFIG_DRM_GEM_DMA_HELPER=y
 CONFIG_DRM_GEM_SHMEM_HELPER=y
 CONFIG_DRM_SUBALLOC_HELPER=m
-CONFIG_DRM_SCHED=m
+CONFIG_DRM_SCHED=y
 
 #
 # Drivers for system framebuffers
@@ -5620,7 +5636,6 @@ CONFIG_DRM_AMD_DC=y
 
 # CONFIG_HSA_AMD is not set
 # CONFIG_DRM_NOUVEAU is not set
-# CONFIG_DRM_XE is not set
 # CONFIG_DRM_VGEM is not set
 # CONFIG_DRM_VKMS is not set
 CONFIG_DRM_ROCKCHIP=y
@@ -5689,10 +5704,12 @@ CONFIG_DRM_PANEL=y
 # CONFIG_DRM_PANEL_HIMAX_HX83112B is not set
 # CONFIG_DRM_PANEL_HIMAX_HX8394 is not set
 # CONFIG_DRM_PANEL_HYDIS_HV101HD1 is not set
+# CONFIG_DRM_PANEL_ILITEK_IL79600A is not set
 # CONFIG_DRM_PANEL_ILITEK_IL9322 is not set
 # CONFIG_DRM_PANEL_ILITEK_ILI9341 is not set
 # CONFIG_DRM_PANEL_ILITEK_ILI9805 is not set
 # CONFIG_DRM_PANEL_ILITEK_ILI9806E is not set
+# CONFIG_DRM_PANEL_ILITEK_ILI9806E_SPI is not set
 # CONFIG_DRM_PANEL_ILITEK_ILI9881C is not set
 # CONFIG_DRM_PANEL_ILITEK_ILI9882T is not set
 # CONFIG_DRM_PANEL_INNOLUX_EJ030NA is not set
@@ -5771,6 +5788,8 @@ CONFIG_DRM_PANEL_SIMPLE=y
 # CONFIG_DRM_PANEL_SUMMIT is not set
 # CONFIG_DRM_PANEL_SYNAPTICS_R63353 is not set
 # CONFIG_DRM_PANEL_TDO_TL070WSH30 is not set
+# CONFIG_DRM_PANEL_TPO_Y17P is not set
+# CONFIG_DRM_PANEL_TDO_DSI_V1 is not set
 # CONFIG_DRM_PANEL_TPO_TD028TTEC1 is not set
 # CONFIG_DRM_PANEL_TPO_TD043MTEA1 is not set
 # CONFIG_DRM_PANEL_TPO_TPG110 is not set
@@ -5780,6 +5799,8 @@ CONFIG_DRM_PANEL_SIMPLE=y
 # CONFIG_DRM_PANEL_VISIONOX_RM69299 is not set
 # CONFIG_DRM_PANEL_VISIONOX_RM692E5 is not set
 # CONFIG_DRM_PANEL_VISIONOX_VTDR6130 is not set
+# CONFIG_DRM_PANEL_WAVESHARE_TOUCHSCREEN is not set
+# CONFIG_DRM_PANEL_WAVESHARE_TOUCHSCREEN_V2 is not set
 # CONFIG_DRM_PANEL_WIDECHIPS_WS2401 is not set
 # CONFIG_DRM_PANEL_XINPENG_XPP055C272 is not set
 # end of Display Panels
@@ -5854,9 +5875,12 @@ CONFIG_DRM_DW_MIPI_DSI=y
 # CONFIG_DRM_IMX8_DC is not set
 # CONFIG_DRM_IMX_DCSS is not set
 # CONFIG_DRM_IMX_LCDC is not set
-CONFIG_DRM_V3D=m
+CONFIG_DRM_V3D=y
 CONFIG_DRM_VC4=m
-# CONFIG_DRM_VC4_HDMI_CEC is not set
+CONFIG_DRM_VC4_HDMI_CEC=y
+# CONFIG_DRM_RP1_DSI is not set
+# CONFIG_DRM_RP1_DPI is not set
+# CONFIG_DRM_RP1_VEC is not set
 # CONFIG_DRM_ETNAVIV is not set
 CONFIG_DRM_HISI_HIBMC=m
 CONFIG_DRM_HISI_KIRIN=y
@@ -5885,7 +5909,7 @@ CONFIG_DRM_MESON_DW_MIPI_DSI=y
 # CONFIG_DRM_XEN_FRONTEND is not set
 # CONFIG_DRM_LIMA is not set
 CONFIG_DRM_PANFROST=m
-CONFIG_DRM_PANTHOR=m
+# CONFIG_DRM_PANTHOR is not set
 # CONFIG_DRM_TIDSS is not set
 # CONFIG_DRM_GUD is not set
 # CONFIG_DRM_ST7571_I2C is not set
@@ -5924,6 +5948,7 @@ CONFIG_BACKLIGHT_CLASS_DEVICE=y
 # CONFIG_BACKLIGHT_KTD2801 is not set
 # CONFIG_BACKLIGHT_KTZ8866 is not set
 # CONFIG_BACKLIGHT_PWM is not set
+# CONFIG_BACKLIGHT_RPI is not set
 # CONFIG_BACKLIGHT_QCOM_WLED is not set
 # CONFIG_BACKLIGHT_ADP8860 is not set
 # CONFIG_BACKLIGHT_ADP8870 is not set
@@ -5961,9 +5986,7 @@ CONFIG_LOGO_LINUX_CLUT224=y
 # CONFIG_TRACE_GPU_MEM is not set
 # end of Graphics support
 
-CONFIG_DRM_ACCEL=y
-CONFIG_DRM_ACCEL_QAIC=m
-CONFIG_DRM_ACCEL_ROCKET=m
+# CONFIG_DRM_ACCEL is not set
 CONFIG_SOUND=m
 CONFIG_SND=m
 CONFIG_SND_TIMER=m
@@ -5992,7 +6015,6 @@ CONFIG_SND_PCM_TIMER=y
 #
 # HD-Audio
 #
-# CONFIG_SND_HDA_INTEL is not set
 # CONFIG_SND_HDA_TEGRA is not set
 # CONFIG_SND_HDA_ACPI is not set
 # end of HD-Audio
@@ -6050,6 +6072,49 @@ CONFIG_SND_SOC_GENERIC_DMAENGINE_PCM=y
 #
 # CONFIG_SND_BCM2835_SOC_I2S is not set
 # CONFIG_SND_BCM63XX_I2S_WHISTLER is not set
+# CONFIG_SND_BCM2708_SOC_CHIPDIP_DAC is not set
+# CONFIG_SND_BCM2708_SOC_GOOGLEVOICEHAT_SOUNDCARD is not set
+# CONFIG_SND_BCM2708_SOC_HIFIBERRY_ADC is not set
+# CONFIG_SND_BCM2708_SOC_HIFIBERRY_ADC8X is not set
+# CONFIG_SND_BCM2708_SOC_HIFIBERRY_DAC is not set
+# CONFIG_SND_BCM2708_SOC_HIFIBERRY_DACPLUS is not set
+# CONFIG_SND_BCM2708_SOC_HIFIBERRY_DACPLUSHD is not set
+# CONFIG_SND_BCM2708_SOC_HIFIBERRY_DACPLUSADC is not set
+# CONFIG_SND_BCM2708_SOC_HIFIBERRY_DACPLUSADCPRO is not set
+# CONFIG_SND_BCM2708_SOC_HIFIBERRY_DACPLUSDSP is not set
+# CONFIG_SND_BCM2708_SOC_HIFIBERRY_STUDIO_DAC8X is not set
+# CONFIG_SND_BCM2708_SOC_HIFIBERRY_DIGI is not set
+# CONFIG_SND_BCM2708_SOC_HIFIBERRY_AMP is not set
+# CONFIG_SND_BCM2708_SOC_PIFI_40 is not set
+# CONFIG_SND_BCM2708_SOC_RPI_CIRRUS is not set
+# CONFIG_SND_BCM2708_SOC_RPI_DAC is not set
+# CONFIG_SND_BCM2708_SOC_RPI_PROTO is not set
+# CONFIG_SND_BCM2708_SOC_JUSTBOOM_BOTH is not set
+# CONFIG_SND_BCM2708_SOC_JUSTBOOM_DAC is not set
+# CONFIG_SND_BCM2708_SOC_JUSTBOOM_DIGI is not set
+# CONFIG_SND_BCM2708_SOC_IQAUDIO_CODEC is not set
+# CONFIG_SND_BCM2708_SOC_IQAUDIO_DAC is not set
+# CONFIG_SND_BCM2708_SOC_IQAUDIO_DIGI is not set
+# CONFIG_SND_BCM2708_SOC_I_SABRE_Q2M is not set
+# CONFIG_SND_BCM2708_SOC_ADAU1977_ADC is not set
+# CONFIG_SND_AUDIOINJECTOR_PI_SOUNDCARD is not set
+# CONFIG_SND_AUDIOINJECTOR_OCTO_SOUNDCARD is not set
+# CONFIG_SND_AUDIOINJECTOR_ISOLATED_SOUNDCARD is not set
+# CONFIG_SND_AUDIOSENSE_PI is not set
+# CONFIG_SND_DIGIDAC1_SOUNDCARD is not set
+# CONFIG_SND_BCM2708_SOC_DIONAUDIO_LOCO is not set
+# CONFIG_SND_BCM2708_SOC_DIONAUDIO_LOCO_V2 is not set
+# CONFIG_SND_BCM2708_SOC_ALLO_PIANO_DAC is not set
+# CONFIG_SND_BCM2708_SOC_ALLO_PIANO_DAC_PLUS is not set
+# CONFIG_SND_BCM2708_SOC_ALLO_BOSS_DAC is not set
+# CONFIG_SND_BCM2708_SOC_ALLO_BOSS2_DAC is not set
+# CONFIG_SND_BCM2708_SOC_ALLO_DIGIONE is not set
+# CONFIG_SND_BCM2708_SOC_ALLO_KATANA_DAC is not set
+# CONFIG_SND_BCM2708_SOC_FE_PI_AUDIO is not set
+# CONFIG_SND_PISOUND is not set
+# CONFIG_SND_RPI_SIMPLE_SOUNDCARD is not set
+# CONFIG_SND_RPI_WM8804_SOUNDCARD is not set
+# CONFIG_SND_DACBERRY400 is not set
 # end of Broadcom
 
 #
@@ -6157,6 +6222,7 @@ CONFIG_SND_SOC_GENERIC_DMAENGINE_PCM=y
 # end of PXA
 
 # CONFIG_SND_SOC_QCOM is not set
+# CONFIG_SND_RP1_AUDIO_OUT is not set
 
 #
 # Renesas
@@ -6241,6 +6307,8 @@ CONFIG_SND_SOC_I2C_AND_SPI=m
 # CODEC drivers
 #
 # CONFIG_SND_SOC_AC97_CODEC is not set
+# CONFIG_SND_SOC_AD193X_SPI is not set
+# CONFIG_SND_SOC_AD193X_I2C is not set
 # CONFIG_SND_SOC_ADAU1372_I2C is not set
 # CONFIG_SND_SOC_ADAU1372_SPI is not set
 # CONFIG_SND_SOC_ADAU1373 is not set
@@ -6322,6 +6390,7 @@ CONFIG_SND_SOC_HDMI_CODEC=m
 # CONFIG_SND_SOC_ICS43432 is not set
 # CONFIG_SND_SOC_IDT821034 is not set
 # CONFIG_SND_SOC_INNO_RK3036 is not set
+# CONFIG_SND_SOC_MA120X0P is not set
 # CONFIG_SND_SOC_MAX98088 is not set
 # CONFIG_SND_SOC_MAX98090 is not set
 # CONFIG_SND_SOC_MAX98357A is not set
@@ -6454,6 +6523,7 @@ CONFIG_SND_SOC_HDMI_CODEC=m
 # CONFIG_SND_SOC_LPASS_VA_MACRO is not set
 # CONFIG_SND_SOC_LPASS_RX_MACRO is not set
 # CONFIG_SND_SOC_LPASS_TX_MACRO is not set
+# CONFIG_SND_SOC_I_SABRE_CODEC is not set
 # end of CODEC drivers
 
 #
@@ -6471,24 +6541,24 @@ CONFIG_HID_SUPPORT=y
 CONFIG_HID=y
 # CONFIG_HID_BATTERY_STRENGTH is not set
 CONFIG_HIDRAW=y
-CONFIG_UHID=m
+# CONFIG_UHID is not set
 CONFIG_HID_GENERIC=y
 # CONFIG_HID_HAPTIC is not set
 
 #
 # Special HID drivers
 #
-CONFIG_HID_A4TECH=m
+CONFIG_HID_A4TECH=y
 # CONFIG_HID_ACCUTOUCH is not set
 # CONFIG_HID_ACRUX is not set
-CONFIG_HID_APPLE=m
+CONFIG_HID_APPLE=y
 # CONFIG_HID_APPLEIR is not set
 # CONFIG_HID_ASUS is not set
 # CONFIG_HID_AUREAL is not set
-CONFIG_HID_BELKIN=m
+CONFIG_HID_BELKIN=y
 # CONFIG_HID_BETOP_FF is not set
 # CONFIG_HID_BIGBEN_FF is not set
-CONFIG_HID_CHERRY=m
+CONFIG_HID_CHERRY=y
 CONFIG_HID_CHICONY=m
 # CONFIG_HID_CORSAIR is not set
 # CONFIG_HID_COUGAR is not set
@@ -6497,14 +6567,14 @@ CONFIG_HID_CHICONY=m
 # CONFIG_HID_CMEDIA is not set
 # CONFIG_HID_CP2112 is not set
 # CONFIG_HID_CREATIVE_SB0540 is not set
-CONFIG_HID_CYPRESS=m
+CONFIG_HID_CYPRESS=y
 # CONFIG_HID_DRAGONRISE is not set
 # CONFIG_HID_EMS_FF is not set
 # CONFIG_HID_ELAN is not set
 # CONFIG_HID_ELECOM is not set
 # CONFIG_HID_ELO is not set
 # CONFIG_HID_EVISION is not set
-CONFIG_HID_EZKEY=m
+CONFIG_HID_EZKEY=y
 # CONFIG_HID_FT260 is not set
 # CONFIG_HID_GEMBIRD is not set
 # CONFIG_HID_GFRM is not set
@@ -6522,12 +6592,12 @@ CONFIG_HID_EZKEY=m
 # CONFIG_HID_VIEWSONIC is not set
 # CONFIG_HID_VRC2 is not set
 # CONFIG_HID_XIAOMI is not set
-CONFIG_HID_GYRATION=m
+CONFIG_HID_GYRATION=y
 # CONFIG_HID_ICADE is not set
-CONFIG_HID_ITE=m
+CONFIG_HID_ITE=y
 # CONFIG_HID_JABRA is not set
 # CONFIG_HID_TWINHAN is not set
-CONFIG_HID_KENSINGTON=m
+CONFIG_HID_KENSINGTON=y
 # CONFIG_HID_LCPOWER is not set
 # CONFIG_HID_LED is not set
 # CONFIG_HID_LENOVO is not set
@@ -6544,17 +6614,17 @@ CONFIG_LOGIWHEELS_FF=y
 # CONFIG_HID_MAYFLASH is not set
 # CONFIG_HID_MEGAWORLD_FF is not set
 CONFIG_HID_REDRAGON=m
-CONFIG_HID_MICROSOFT=m
-CONFIG_HID_MONTEREY=m
-CONFIG_HID_MULTITOUCH=m
+CONFIG_HID_MICROSOFT=y
+CONFIG_HID_MONTEREY=y
+# CONFIG_HID_MULTITOUCH is not set
 # CONFIG_HID_NINTENDO is not set
 # CONFIG_HID_NTI is not set
 # CONFIG_HID_NTRIG is not set
 # CONFIG_HID_ORTEK is not set
-CONFIG_HID_PANTHERLORD=m
+CONFIG_HID_PANTHERLORD=y
 CONFIG_PANTHERLORD_FF=y
 # CONFIG_HID_PENMOUNT is not set
-CONFIG_HID_PETALYNX=m
+CONFIG_HID_PETALYNX=y
 # CONFIG_HID_PICOLCD is not set
 # CONFIG_HID_PLANTRONICS is not set
 # CONFIG_HID_PLAYSTATION is not set
@@ -6571,13 +6641,13 @@ CONFIG_HID_SAMSUNG=m
 # CONFIG_HID_SPEEDLINK is not set
 # CONFIG_HID_STEAM is not set
 # CONFIG_HID_STEELSERIES is not set
-CONFIG_HID_SUNPLUS=m
+CONFIG_HID_SUNPLUS=y
 # CONFIG_HID_RMI is not set
 # CONFIG_HID_GREENASIA is not set
 # CONFIG_HID_HYPERV_MOUSE is not set
 # CONFIG_HID_SMARTJOYPLUS is not set
 # CONFIG_HID_TIVO is not set
-CONFIG_HID_TOPSEED=m
+CONFIG_HID_TOPSEED=y
 # CONFIG_HID_TOPRE is not set
 # CONFIG_HID_THINGM is not set
 # CONFIG_HID_THRUSTMASTER is not set
@@ -6683,6 +6753,7 @@ CONFIG_USB_OHCI_HCD_PLATFORM=y
 # CONFIG_USB_SL811_HCD is not set
 # CONFIG_USB_R8A66597_HCD is not set
 # CONFIG_USB_RENESAS_USBHS_HCD is not set
+# CONFIG_USB_DWCOTG is not set
 # CONFIG_USB_HCD_TEST_MODE is not set
 # CONFIG_USB_XEN_HCD is not set
 CONFIG_USB_RENESAS_USBHS=y
@@ -6792,13 +6863,13 @@ CONFIG_USB_SERIAL_GENERIC=y
 # CONFIG_USB_SERIAL_AIRCABLE is not set
 # CONFIG_USB_SERIAL_ARK3116 is not set
 # CONFIG_USB_SERIAL_BELKIN is not set
-CONFIG_USB_SERIAL_CH341=m
+CONFIG_USB_SERIAL_CH341=y
 # CONFIG_USB_SERIAL_WHITEHEAT is not set
 # CONFIG_USB_SERIAL_DIGI_ACCELEPORT is not set
-CONFIG_USB_SERIAL_CP210X=m
+CONFIG_USB_SERIAL_CP210X=y
 # CONFIG_USB_SERIAL_CYPRESS_M8 is not set
 # CONFIG_USB_SERIAL_EMPEG is not set
-CONFIG_USB_SERIAL_FTDI_SIO=m
+CONFIG_USB_SERIAL_FTDI_SIO=y
 # CONFIG_USB_SERIAL_VISOR is not set
 # CONFIG_USB_SERIAL_IPAQ is not set
 # CONFIG_USB_SERIAL_IR is not set
@@ -6819,7 +6890,7 @@ CONFIG_USB_SERIAL_FTDI_SIO=m
 # CONFIG_USB_SERIAL_MOS7840 is not set
 # CONFIG_USB_SERIAL_MXUPORT is not set
 # CONFIG_USB_SERIAL_NAVMAN is not set
-CONFIG_USB_SERIAL_PL2303=m
+CONFIG_USB_SERIAL_PL2303=y
 # CONFIG_USB_SERIAL_OTI6858 is not set
 # CONFIG_USB_SERIAL_QCAUX is not set
 # CONFIG_USB_SERIAL_QUALCOMM is not set
@@ -7022,31 +7093,32 @@ CONFIG_MMC_BLOCK_MINORS=32
 #
 # MMC/SD/SDIO Host Controller Drivers
 #
+# CONFIG_MMC_BCM2835_MMC is not set
 # CONFIG_MMC_DEBUG is not set
 CONFIG_MMC_ARMMMCI=y
 CONFIG_MMC_QCOM_DML=y
 CONFIG_MMC_STM32_SDMMC=y
-CONFIG_MMC_SDHCI=m
+CONFIG_MMC_SDHCI=y
 CONFIG_MMC_SDHCI_IO_ACCESSORS=y
-CONFIG_MMC_SDHCI_UHS2=m
-CONFIG_MMC_SDHCI_PCI=m
+CONFIG_MMC_SDHCI_UHS2=y
+CONFIG_MMC_SDHCI_PCI=y
 CONFIG_MMC_RICOH_MMC=y
-CONFIG_MMC_SDHCI_ACPI=m
-CONFIG_MMC_SDHCI_PLTFM=m
-CONFIG_MMC_SDHCI_OF_ARASAN=m
+CONFIG_MMC_SDHCI_ACPI=y
+CONFIG_MMC_SDHCI_PLTFM=y
+CONFIG_MMC_SDHCI_OF_ARASAN=y
 # CONFIG_MMC_SDHCI_OF_AT91 is not set
-CONFIG_MMC_SDHCI_OF_ESDHC=m
-CONFIG_MMC_SDHCI_OF_DWCMSHC=m
-CONFIG_MMC_SDHCI_CADENCE=m
+CONFIG_MMC_SDHCI_OF_ESDHC=y
+CONFIG_MMC_SDHCI_OF_DWCMSHC=y
+CONFIG_MMC_SDHCI_CADENCE=y
 # CONFIG_MMC_SDHCI_ESDHC_IMX is not set
-CONFIG_MMC_SDHCI_TEGRA=m
+CONFIG_MMC_SDHCI_TEGRA=y
 # CONFIG_MMC_SDHCI_PXAV3 is not set
-CONFIG_MMC_SDHCI_F_SDH30=m
+CONFIG_MMC_SDHCI_F_SDH30=y
 # CONFIG_MMC_SDHCI_MILBEAUT is not set
-CONFIG_MMC_SDHCI_IPROC=m
+CONFIG_MMC_SDHCI_IPROC=y
 CONFIG_MMC_MESON_GX=y
 # CONFIG_MMC_MESON_MX_SDIO is not set
-CONFIG_MMC_SDHCI_MSM=m
+CONFIG_MMC_SDHCI_MSM=y
 # CONFIG_MMC_MXC is not set
 # CONFIG_MMC_TIFM_SD is not set
 CONFIG_MMC_SPI=y
@@ -7076,8 +7148,8 @@ CONFIG_MMC_CQHCI=y
 # CONFIG_MMC_TOSHIBA_PCI is not set
 CONFIG_MMC_BCM2835=y
 # CONFIG_MMC_MTK is not set
-CONFIG_MMC_SDHCI_BRCMSTB=m
-CONFIG_MMC_SDHCI_XENON=m
+CONFIG_MMC_SDHCI_BRCMSTB=y
+CONFIG_MMC_SDHCI_XENON=y
 # CONFIG_MMC_SDHCI_AM654 is not set
 # CONFIG_MMC_OWL is not set
 CONFIG_SCSI_UFSHCD=m
@@ -7176,30 +7248,27 @@ CONFIG_LEDS_TRIGGER_DEFAULT_ON=y
 #
 # CONFIG_LEDS_TRIGGER_TRANSIENT is not set
 # CONFIG_LEDS_TRIGGER_CAMERA is not set
+# CONFIG_LEDS_TRIGGER_INPUT is not set
 # CONFIG_LEDS_TRIGGER_PANIC is not set
 # CONFIG_LEDS_TRIGGER_NETDEV is not set
 # CONFIG_LEDS_TRIGGER_PATTERN is not set
 # CONFIG_LEDS_TRIGGER_TTY is not set
 # CONFIG_LEDS_TRIGGER_INPUT_EVENTS is not set
+# CONFIG_LEDS_TRIGGER_ACTPWR is not set
 
 #
 # Simatic LED drivers
 #
 # CONFIG_ACCESSIBILITY is not set
 CONFIG_INFINIBAND=m
-CONFIG_INFINIBAND_USER_MAD=m
-CONFIG_INFINIBAND_USER_ACCESS=m
-CONFIG_INFINIBAND_USER_MEM=y
-CONFIG_INFINIBAND_ON_DEMAND_PAGING=y
+# CONFIG_INFINIBAND_USER_MAD is not set
+# CONFIG_INFINIBAND_USER_ACCESS is not set
 CONFIG_INFINIBAND_ADDR_TRANS=y
 CONFIG_INFINIBAND_ADDR_TRANS_CONFIGFS=y
 CONFIG_INFINIBAND_VIRT_DMA=y
 CONFIG_INFINIBAND_BNXT_RE=m
 # CONFIG_INFINIBAND_CXGB4 is not set
-# CONFIG_INFINIBAND_EFA is not set
-# CONFIG_INFINIBAND_ERDMA is not set
 CONFIG_INFINIBAND_HNS_HIP08=m
-CONFIG_INFINIBAND_IRDMA=m
 CONFIG_MLX4_INFINIBAND=m
 CONFIG_MLX5_INFINIBAND=m
 # CONFIG_INFINIBAND_MTHCA is not set
@@ -7257,6 +7326,7 @@ CONFIG_RTC_INTF_DEV=y
 # CONFIG_RTC_DRV_ABB5ZES3 is not set
 # CONFIG_RTC_DRV_ABEOZ9 is not set
 # CONFIG_RTC_DRV_ABX80X is not set
+CONFIG_RTC_DRV_RPI=y
 CONFIG_RTC_DRV_BRCMSTB=y
 CONFIG_RTC_DRV_DS1307=y
 # CONFIG_RTC_DRV_DS1307_CENTURY is not set
@@ -7409,6 +7479,7 @@ CONFIG_PL330_DMA=y
 CONFIG_TEGRA186_GPC_DMA=y
 CONFIG_TEGRA20_APB_DMA=y
 CONFIG_TEGRA210_ADMA=y
+CONFIG_DMA_BCM2708=y
 # CONFIG_XGENE_DMA is not set
 # CONFIG_XILINX_DMA is not set
 # CONFIG_XILINX_XDMA is not set
@@ -7501,17 +7572,17 @@ CONFIG_VMGENID=m
 # CONFIG_ARM_CCA_GUEST is not set
 CONFIG_VIRTIO_ANCHOR=y
 CONFIG_VIRTIO=y
-CONFIG_VIRTIO_PCI_LIB=m
-CONFIG_VIRTIO_PCI_LIB_LEGACY=m
+CONFIG_VIRTIO_PCI_LIB=y
+CONFIG_VIRTIO_PCI_LIB_LEGACY=y
 CONFIG_VIRTIO_MENU=y
-CONFIG_VIRTIO_PCI=m
+CONFIG_VIRTIO_PCI=y
 CONFIG_VIRTIO_PCI_LEGACY=y
 CONFIG_VIRTIO_VDPA=m
 CONFIG_VIRTIO_PMEM=y
-CONFIG_VIRTIO_BALLOON=m
+CONFIG_VIRTIO_BALLOON=y
 # CONFIG_VIRTIO_MEM is not set
-CONFIG_VIRTIO_INPUT=m
-CONFIG_VIRTIO_MMIO=m
+CONFIG_VIRTIO_INPUT=y
+CONFIG_VIRTIO_MMIO=y
 CONFIG_VIRTIO_MMIO_CMDLINE_DEVICES=y
 CONFIG_VIRTIO_DMA_SHARED_BUFFER=m
 # CONFIG_VIRTIO_DEBUG is not set
@@ -7548,7 +7619,6 @@ CONFIG_HYPERV=y
 CONFIG_HYPERV_UTILS=m
 CONFIG_HYPERV_BALLOON=m
 CONFIG_HYPERV_VMBUS=y
-# CONFIG_MSHV_ROOT is not set
 # end of Microsoft Hyper-V guest support
 
 #
@@ -7609,6 +7679,7 @@ CONFIG_COMMON_CLK=y
 # CONFIG_COMMON_CLK_MAX9485 is not set
 CONFIG_COMMON_CLK_RK808=y
 CONFIG_COMMON_CLK_RP1=y
+CONFIG_COMMON_CLK_RP1_SDIO=y
 CONFIG_COMMON_CLK_HI655X=y
 CONFIG_COMMON_CLK_SCMI=y
 CONFIG_COMMON_CLK_SCPI=y
@@ -8092,6 +8163,7 @@ CONFIG_SUN6I_MSGBOX=y
 # CONFIG_QCOM_CPUCP_MBOX is not set
 # CONFIG_QCOM_IPCC is not set
 CONFIG_BCM74110_MAILBOX=y
+CONFIG_MBOX_RP1=y
 CONFIG_IOMMU_IOVA=y
 CONFIG_IOMMU_API=y
 CONFIG_IOMMUFD_DRIVER=y
@@ -8133,6 +8205,7 @@ CONFIG_TEGRA_IOMMU_SMMU=y
 # CONFIG_MTK_IOMMU is not set
 # CONFIG_VIRTIO_IOMMU is not set
 # CONFIG_SPRD_IOMMU is not set
+CONFIG_BCM2712_IOMMU=y
 
 #
 # Remoteproc drivers
@@ -8315,7 +8388,6 @@ CONFIG_XLNX_EVENT_MANAGER=y
 # Amlogic PM Domains
 #
 CONFIG_MESON_EE_PM_DOMAINS=y
-CONFIG_MESON_SECURE_PM_DOMAINS=y
 # end of Amlogic PM Domains
 
 CONFIG_ARM_SCMI_PERF_DOMAIN=y
@@ -8342,6 +8414,7 @@ CONFIG_IMX9_BLK_CTRL=y
 #
 CONFIG_MTK_SCPSYS=y
 CONFIG_MTK_SCPSYS_PM_DOMAINS=y
+# CONFIG_AIROHA_CPU_PM_DOMAIN is not set
 # end of MediaTek PM Domains
 
 #
@@ -9041,7 +9114,7 @@ CONFIG_PWM=y
 CONFIG_PWM_BCM2835=y
 CONFIG_PWM_BCM_IPROC=y
 # CONFIG_PWM_BERLIN is not set
-# CONFIG_PWM_BRCMSTB is not set
+CONFIG_PWM_BRCMSTB=y
 # CONFIG_PWM_CLK is not set
 # CONFIG_PWM_DWC is not set
 # CONFIG_PWM_FSL_FTM is not set
@@ -9056,9 +9129,11 @@ CONFIG_PWM_MESON=y
 # CONFIG_PWM_MTK_DISP is not set
 # CONFIG_PWM_OMAP_DMTIMER is not set
 # CONFIG_PWM_PCA9685 is not set
+CONFIG_PWM_PIO_RP1=y
 CONFIG_PWM_RASPBERRYPI_POE=y
 # CONFIG_PWM_RENESAS_RCAR is not set
 # CONFIG_PWM_RENESAS_TPU is not set
+CONFIG_PWM_RP1=y
 CONFIG_PWM_ROCKCHIP=y
 # CONFIG_PWM_SPRD is not set
 CONFIG_PWM_SUN4I=y
@@ -9265,10 +9340,7 @@ CONFIG_PHY_TEGRA194_P2U=y
 # CONFIG_PHY_XILINX_ZYNQMP is not set
 # end of PHY Subsystem
 
-CONFIG_POWERCAP=y
-# CONFIG_IDLE_INJECT is not set
-# CONFIG_ARM_SCMI_POWERCAP is not set
-# CONFIG_DTPM is not set
+# CONFIG_POWERCAP is not set
 # CONFIG_MCB is not set
 
 #
@@ -9295,6 +9367,7 @@ CONFIG_ARM64_BRBE=y
 # CONFIG_ARM_DMC620_PMU is not set
 # CONFIG_MARVELL_CN10K_TAD_PMU is not set
 # CONFIG_ALIBABA_UNCORE_DRW_PMU is not set
+# CONFIG_RPI_AXIPERF is not set
 CONFIG_HISI_PMU=m
 CONFIG_HISI_PCIE_PMU=m
 CONFIG_HNS3_PMU=m
@@ -9340,12 +9413,12 @@ CONFIG_NVMEM_LAYOUTS=y
 # CONFIG_NVMEM_LAYOUT_U_BOOT_ENV is not set
 # end of Layout Types
 
+CONFIG_NVMEM_RASPBERRYPI_OTP=y
 CONFIG_NVMEM_BCM_OCOTP=y
 # CONFIG_NVMEM_IMX_IIM is not set
 # CONFIG_NVMEM_IMX_OCOTP is not set
 # CONFIG_NVMEM_IMX_OCOTP_ELE is not set
 # CONFIG_NVMEM_LAYERSCAPE_SFP is not set
-# CONFIG_NVMEM_MESON_EFUSE is not set
 # CONFIG_NVMEM_MESON_MX_EFUSE is not set
 # CONFIG_NVMEM_MTK_EFUSE is not set
 # CONFIG_NVMEM_QCOM_QFPROM is not set
@@ -9761,8 +9834,30 @@ CONFIG_INTEGRITY_TRUSTED_KEYRING=y
 CONFIG_INTEGRITY_PLATFORM_KEYRING=y
 CONFIG_LOAD_UEFI_KEYS=y
 CONFIG_INTEGRITY_AUDIT=y
-# CONFIG_IMA is not set
-# CONFIG_IMA_SECURE_AND_OR_TRUSTED_BOOT is not set
+CONFIG_IMA=y
+# CONFIG_IMA_KEXEC is not set
+CONFIG_IMA_MEASURE_PCR_IDX=10
+CONFIG_IMA_LSM_RULES=y
+CONFIG_IMA_NG_TEMPLATE=y
+# CONFIG_IMA_SIG_TEMPLATE is not set
+CONFIG_IMA_DEFAULT_TEMPLATE="ima-ng"
+# CONFIG_IMA_DEFAULT_HASH_SHA1 is not set
+# CONFIG_IMA_DEFAULT_HASH_SHA256 is not set
+CONFIG_IMA_DEFAULT_HASH_SHA512=y
+CONFIG_IMA_DEFAULT_HASH="sha512"
+CONFIG_IMA_WRITE_POLICY=y
+CONFIG_IMA_READ_POLICY=y
+CONFIG_IMA_APPRAISE=y
+CONFIG_IMA_ARCH_POLICY=y
+# CONFIG_IMA_APPRAISE_BUILD_POLICY is not set
+CONFIG_IMA_APPRAISE_BOOTPARAM=y
+# CONFIG_IMA_APPRAISE_MODSIG is not set
+# CONFIG_IMA_BLACKLIST_KEYRING is not set
+# CONFIG_IMA_LOAD_X509 is not set
+CONFIG_IMA_MEASURE_ASYMMETRIC_KEYS=y
+CONFIG_IMA_QUEUE_EARLY_BOOT_KEYS=y
+CONFIG_IMA_SECURE_AND_OR_TRUSTED_BOOT=y
+# CONFIG_IMA_DISABLE_HTABLE is not set
 # CONFIG_EVM is not set
 # CONFIG_DEFAULT_SECURITY_SELINUX is not set
 # CONFIG_DEFAULT_SECURITY_APPARMOR is not set
@@ -10008,13 +10103,16 @@ CONFIG_CRYPTO_AES_ARM64_CE_CCM=y
 CONFIG_CRYPTO_HW=y
 CONFIG_CRYPTO_DEV_ALLWINNER=y
 CONFIG_CRYPTO_DEV_SUN4I_SS=y
+# CONFIG_CRYPTO_DEV_SUN4I_SS_PRNG is not set
 # CONFIG_CRYPTO_DEV_SUN4I_SS_DEBUG is not set
 CONFIG_CRYPTO_DEV_SUN8I_CE=y
 # CONFIG_CRYPTO_DEV_SUN8I_CE_DEBUG is not set
 # CONFIG_CRYPTO_DEV_SUN8I_CE_HASH is not set
+# CONFIG_CRYPTO_DEV_SUN8I_CE_PRNG is not set
 # CONFIG_CRYPTO_DEV_SUN8I_CE_TRNG is not set
 CONFIG_CRYPTO_DEV_SUN8I_SS=y
 # CONFIG_CRYPTO_DEV_SUN8I_SS_DEBUG is not set
+# CONFIG_CRYPTO_DEV_SUN8I_SS_PRNG is not set
 # CONFIG_CRYPTO_DEV_SUN8I_SS_HASH is not set
 # CONFIG_CRYPTO_DEV_FSL_CAAM is not set
 # CONFIG_CRYPTO_DEV_SAHARA is not set
@@ -10205,7 +10303,6 @@ CONFIG_ARCH_HAS_DMA_PREP_COHERENT=y
 CONFIG_ARCH_HAS_FORCE_DMA_UNENCRYPTED=y
 CONFIG_SWIOTLB=y
 # CONFIG_SWIOTLB_DYNAMIC is not set
-CONFIG_DMA_BOUNCE_UNALIGNED_KMALLOC=y
 CONFIG_DMA_NEED_SYNC=y
 # CONFIG_DMA_RESTRICTED_POOL is not set
 CONFIG_DMA_NONCOHERENT_MMAP=y
@@ -10265,6 +10362,7 @@ CONFIG_MEMREGION=y
 CONFIG_ARCH_HAS_UACCESS_FLUSHCACHE=y
 CONFIG_ARCH_STACKWALK=y
 CONFIG_STACKDEPOT=y
+CONFIG_STACKDEPOT_ALWAYS_INIT=y
 CONFIG_STACKDEPOT_MAX_FRAMES=64
 CONFIG_SBITMAP=y
 CONFIG_PARMAN=m
@@ -10360,7 +10458,6 @@ CONFIG_UBSAN_ENUM=y
 # CONFIG_UBSAN_KVM_EL2 is not set
 CONFIG_HAVE_ARCH_KCSAN=y
 CONFIG_HAVE_KCSAN_COMPILER=y
-# CONFIG_KCSAN is not set
 # end of Generic Kernel Debugging Instruments
 
 #
@@ -10379,6 +10476,7 @@ CONFIG_PAGE_EXTENSION=y
 # CONFIG_DEBUG_PAGEALLOC is not set
 CONFIG_SLUB_DEBUG=y
 # CONFIG_SLUB_DEBUG_ON is not set
+# CONFIG_SLUB_RCU_DEBUG is not set
 # CONFIG_PAGE_OWNER is not set
 CONFIG_PAGE_TABLE_CHECK=y
 CONFIG_PAGE_TABLE_CHECK_ENFORCED=y
@@ -10413,7 +10511,12 @@ CONFIG_HAVE_ARCH_KASAN_VMALLOC=y
 CONFIG_CC_HAS_KASAN_GENERIC=y
 CONFIG_CC_HAS_KASAN_SW_TAGS=y
 CONFIG_CC_HAS_WORKING_NOSANITIZE_ADDRESS=y
-# CONFIG_KASAN is not set
+CONFIG_KASAN=y
+# CONFIG_KASAN_GENERIC is not set
+# CONFIG_KASAN_SW_TAGS is not set
+CONFIG_KASAN_HW_TAGS=y
+CONFIG_KASAN_VMALLOC=y
+# CONFIG_KASAN_EXTRA_INFO is not set
 CONFIG_HAVE_ARCH_KFENCE=y
 CONFIG_KFENCE=y
 CONFIG_KFENCE_SAMPLE_INTERVAL=100
diff --git a/kernel/prepare/pkg.yaml b/kernel/prepare/pkg.yaml
index 6c6e61d..27c599b 100644
--- a/kernel/prepare/pkg.yaml
+++ b/kernel/prepare/pkg.yaml
@@ -6,8 +6,8 @@ dependencies:
   - image: "{{ .LLVM_IMAGE }}:{{ .TOOLS_REV }}"
 steps:
   - sources:
-      - url: https://cdn.kernel.org/pub/linux/kernel/v{{ regexReplaceAll "(\\d+)(.\\d+)(\\.\\d+)?$" .linux_version "${1}" }}.x/linux-{{ .linux_version }}.tar.xz
-        destination: linux.tar.xz
+      - url: "https://github.com/raspberrypi/linux/archive/{{ .linux_version }}.tar.gz"
+        destination: linux.tar.gz
         sha256: "{{ .linux_sha256 }}"
         sha512: "{{ .linux_sha512 }}"
       - url: https://github.com/a13xp0p0v/kernel-hardening-checker/archive/{{ .kspp_ref }}.tar.gz
@@ -19,7 +19,7 @@ steps:
     prepare:
       - |
         mkdir -p /src
-        tar -xJf linux.tar.xz --strip-components=1 -C /src
+        tar -xzf linux.tar.gz --strip-components=1 -C /src
 
         cd /src
         make mrproper
-- 
2.53.0

